File: airesources/Go/MyBot.go
package main

import (
	"fmt"
	"./src/hlt"
	"log"
	"os"
	"strconv"
)
// golang starter kit with logging and basic pathfinding
// Arjun Viswanathan 2017 / github arjunvis

func main() {
	logging := true
	botName := "GoBot"

	conn := hlt.NewConnection(botName)

	// set up logging
	if logging {
		fname := strconv.Itoa(conn.PlayerTag) + "_gamelog.log"
		f, err := os.OpenFile(fname, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
		if err != nil {
			fmt.Println("error opening file: %v", err)
		}
		defer f.Close()
		log.SetOutput(f)
	}
	gameMap := conn.UpdateMap()
	gameturn := 1
	for true {
		gameMap = conn.UpdateMap()
		commandQueue := [] string{}

		myPlayer := gameMap.Players[gameMap.MyId]
		myShips := myPlayer.Ships

		for i := 0; i < len(myShips); i++ {
			ship := myShips[i]
			if ship.DockingStatus == hlt.UNDOCKED {
				commandQueue = append(commandQueue, hlt.StrategyBasicBot(ship, gameMap))
			}
		}
		log.Printf("Turn %v\n",gameturn)
		conn.SubmitCommands(commandQueue)
		gameturn++
	}
}

File: airesources/FSharp/Halite2/MyBot.fs
open System
open hlt

[<EntryPoint>]
let main argv =
    let botName = 
        match argv.Length with 
            | x when x > 0 -> argv.[0] 
            | _ -> "Ephesus"

    let conn = newConnection botName

    let mutable gameMap = updateMap conn
    let mutable gameTurn = 1

    while true do
        gameMap <- updateMap conn

        let myPlayer = 
            gameMap.Players 
                |> Array.find (fun p -> p.Id = gameMap.MyId)
        
        let commandQueue = 
            myPlayer.Ships
                |> Array.filter (fun s -> s.DockingStatus = Undocked)
                |> Array.map (fun s -> strategyBasicBot s gameMap)

        submitCommands commandQueue

        gameTurn <- gameTurn + 1      

    0 // return an integer exit code
